### PR TITLE
Add a colorful version of the mini colorscheme

### PR DESCRIPTION
**Pull request by Feuerfuchs**
Unfortunately, I couldn't find a good name for the option. :/

### CODE REVIEW

File: sol.c [hunk 1]
```diff
@@ -70,7 +70,8 @@ int main(int argc, char** argv) {
 #endif
 		case 'b': op.s = &unicode_large_mono; break;
 		case 'c': op.s = &unicode_large_color; break;
-		case 'm': op.s = &unicode_small_mono; break; /* "mini" */
+		case 'm': op.s = &unicode_small_mono; break; /* "mini, bland" */
+		case 'n': op.s = &unicode_small_color; break; /* "mini, colorful" */
```
Comment (girst) [inline sol.c]: can you change the option to '-M' please?

File: sol.h [hunk 2]
```diff
@@ -195,7 +195,8 @@ struct undo undo_sentinel;
 	LONGHELP_SPECIFIC \
 	"    -b(land colorscheme)\n" \
 	"    -c(olorful colorscheme)\n" \
-	"    -m(iniature colorscheme)\n" \
+	"    -m(iniature colorscheme, bland)\n" \
```
Comment (girst) [inline sol.h]: call it 'monochrome' instead of bland please. that name was just a hack, so it starts with the correct letter.

**Review by girst — CHANGES_REQUESTED** [review]
thanks, that looks like a neat addition! i've left 2 comments on the `diff` page ;-)

File: sol.h (hunk 2 above)
Comment (girst) [reply sol.h]: that applies to the getopt part as well

File: schemes.h [hunk 3]
```diff
@@ -26,6 +26,8 @@ struct scheme {
 #define BULCARD(s, r) \
 	ULCARD("\017\033[2m" s "\033[22m", r)//WARN: requires hack in print_hi()
 #define USCARD(c) (char*[]){c}
+#define RUSCARD(c) (char*[]){"\033[91m" c "\033[39m"}
+#define BUSCARD(c) (char*[]){c}
```
Comment (girst) [inline schemes.h]: these macros must have the same number of bytes before and including the card character. (spider's foundation prints `halfcard[2]` number of bytes)

currently, halfcard2 is 4 bytes, i.e. exactly the utf8-sequence for the char, but no colour sequence. pad buscard with `\017` 5 times before the `c` and increase halfcard2 by 5 to fix this

Comment (girst) [reply schemes.h]: on second thought, padding should be unnecessary (as it stops at the null terminator anyways). but change `halfwidth` to `{1,1,9}`

Comment (Feuerfuchs) [reply schemes.h]: I see, I was wondering why that `\017` is in the `ULCARD` macro. Makes sense now!

Just to clarify, by `halfcard2` you mean index 2 of `halfwidth`, right?

Comment (Feuerfuchs) [reply schemes.h]: Ah, your second comment wasn't there when I wrote my comment. ^^

Comment (girst) [reply schemes.h]: argh! my comment left by email was apparently dropped. yes, all instances of halfcard should have been `halfwidth[2]`, sorry for any confusion. i think we can do without the padding :)

Comment (Feuerfuchs) [reply schemes.h]: Done! :D

**Comment by girst** [comment]
ok, github just ate my email again. 

```
just tested it and I've instructed you to do the wrong thing (sorry). using halfwidth[2] as 9 chops off the colour reset sequence! all cards are then tinted red.

the BUSCARD and RUSCARD macros should include the space right after the colour-reset.sequence. when you're using them in lines 195++ you should remove the space there. length of halfwidth[2] should then be 14, if i have counted correctly

I'm sorry. it only shows that the scheme structure is all but optimal :/

btw. I've tested that with gdb. attach it with `gdb --pid $(pidof spider)` and then type `print f.f[0][RANK_K] = HEA_K` (without all the backticks) to simulate a completed pile. then `continue` to update the screen
```

**Comment by Feuerfuchs** [comment]
I tried the change you suggested and unfortunately, it broke the game:
bildschirmfoto vom 2019-02-14 20-17-43

So I finally took a closer look at the code to try to understand how it works. Turns out that the space character after the card characters messed the calculations up. So whenever a half red card was supposed to be displayed, the cutoff occurred here (☐ = placeholder for any card character):
```
"\033[91m☐ \033[39m"
▕ ---------------▕
```
The m got removed and so the control sequence didn't stop.

Another issue was that uncolored half cards were displayed as full cards since we simply let the null terminator cut the string off. That retained the space character after the card character.

```
"☐ "
▕ ▕
```

I now adjusted the code so the space character is placed in the macro itself *after* the control sequence that stops the red color. Also, I applied padding to uncolored cards to fix the alignment issues. :)

**Comment by girst** [comment]
great! this now works perfectly! I'll merge this later this evening, after cleaning up my local tree ;)

yeah, that hack with half cards was proooobaaablyyyyy not so smart in hindsight. cudos for figuring out my clusterfuck of a print function :D

**Review by girst — APPROVED** [review]

**Comment by Feuerfuchs** [comment]
I just now realized you explained that white space fix to me in your previous message:

> the BUSCARD and RUSCARD macros should include the space right after the colour-reset.sequence. when you're using them in lines 195++ you should remove the space there. length of halfwidth[2] should then be 14, if i have counted correctly

Somehow, only the last part registered. I have no idea why I'm such a mess today. xD

**Comment by girst** [comment]
merged as 6a597bc -- thank you very much!

**Pull request closed without merging**